syntax = "proto3";

package server_commands;

enum Command {
  UNKNOWN = 0;
  START = 1;
  STOP = 2;
  RESTART = 3;
}

enum FrameOrientation {
  UNKNOWN_ORIENTATION = 0;
  PORTRAIT = 1;
  LANDSCAPE = 2;
}

enum Rotation {
  UNKNOWN_ROTATION = 0;
  ROTATE_0 = 1;
  ROTATE_90 = 2;
  ROTATE_180 = 3;
  ROTATE_270 = 4;
}

enum ProcessorType {
  ANY = 0;
  GPU = 1;
  CPU = 2;
}

service ServerCommands {
  rpc ExecuteCommand(ExecuteCommandRequest) returns (ExecuteCommandResponse);
}

message ExecuteCommandRequest {
  Command command = 1;
  string name = 2;
  string call_back_url = 3;
  string input_url = 4;
  FrameOrientation frame_orientation = 5;
  Rotation rotation = 6;
  ProcessorType processor_type = 7;
}

message ExecuteCommandResponse {
  bool success = 1;
  optional string message = 2;
}